### PR TITLE
Added missing pagination properties

### LINKED ISSUE #2333: [Bug]: Missing pagination property in dab.draft.schema.json
### What happened?

I added the pagination section in the runtime section of the config file as following:

`"pagination": {
            "max-page-size": -1, 
            "default-page-size": 100 
          }`

`dab start` works fine, but to generate the merged config file I use `dab validate` and this fails, because the "pagination" section is missing in the dab.draft.schema.json file

### Version

1.1.7

### What database are you using?

Azure SQL

### What hosting model are you using?

Local (including CLI), Container Apps

### Which API approach are you accessing DAB through?

GraphQL

### Relevant log output

```Text
...
Error: > Total schema validation errors: 1
> NoAdditionalPropertiesAllowed: #/runtime.pagination at 41:17


Error: Config is invalid. Check above logs for details.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by JerryNixon**
Closes #2333

## Overview

```json
{
  "runtime": {
    "host": {
      "max-response-size-mb": 158
    },
    "pagination": {
      "max-page-size": 100000,
      "default-page-size": 100,
    }
  }
}
```

## `max-response-size-mb`

[source](https://github.com/Azure/data-api-builder/blob/b7c304d84cf5d4aefd176220ddd7939edc49d759/src/Config/ObjectModel/PaginationOptions.cs#L37)

*Specifies the maximum size, in megabytes, of the database response allowed in a single result. This limit helps prevent large results (in size) from overwhelming the API.*

```json
{
  "runtime": {
    "host": {
      "max-response-size-mb": 158
    }
  }
}
```

| Min | Max    | Special | Optional | Default |
|-----|--------|---------|----------|---------|
| 1 MB| 158 MB | N/A     | Yes      | 158 MB  |

---

## `max-page-size`

[source](https://github.com/Azure/data-api-builder/blob/b7c304d84cf5d4aefd176220ddd7939edc49d759/src/Config/ObjectModel/PaginationOptions.cs#L37)

*Defines the maximum number of records that can be returned in a single page of results. This limit helps prevent large results (in rows) from overwhelming the API.*

```json
{
  "runtime": {
    "pagination": {
      "max-page-size": 100000
    }
  }
}
```

| Min | Max    | Special             | Optional | Default |
|-----|--------|---------------------|----------|---------|
| 1   | 100,000| `-1` (unlimited)     | Yes      | 100,000 |

---

## `default-page-size`

[source](https://github.com/Azure/data-api-builder/blob/b7c304d84cf5d4aefd176220ddd7939edc49d759/src/Config/ObjectModel/PaginationOptions.cs#L31)

*Sets the default number of records returned in a single response when the result set exceeds this limit. When this limit is reached, a continuation token is provided to help users retrieve the next page.*

```json
{
  "runtime": {
    "pagination": {
      "default-page-size": 100
    }
  }
}
```

| Min | Max    | Special                              | Optional | Default |
|-----|--------|--------------------------------------|----------|---------|
| 1   | 100,000| `-1` (same as `max-page-size`)        | Yes      | 100     |

### CODE REVIEW

**Comment by abhishekkumams** [comment]
can user specify any of these properties as `null`?

for example:
Is this valid?
```
{
  "runtime": {
    "pagination": {
      "default-page-size": null
    }
  }
}
```

File: schemas/dab.draft.schema.json [hunk 1]
```diff
@@ -806,4 +833,4 @@
       ]
     }
   }
-}
+}
```
Comment (seantleonard) [inline schemas/dab.draft.schema.json:836]: missing new line at end of file

File: schemas/dab.draft.schema.json [hunk 2]
```diff
@@ -207,6 +227,13 @@
           "description": "Global hosting configuration",
           "additionalProperties": false,
           "properties": {
+            "max-response-size-mb": {
+              "type": "integer",
```
Comment (seantleonard) [inline schemas/dab.draft.schema.json:231]:
```suggestion
             "type": [
                "integer",
                "null"
              ]
```

as this isn't a required field in the config.

File: schemas/dab.draft.schema.json [hunk 3]
```diff
@@ -139,6 +139,23 @@
       "description": "Runtime configuration settings",
       "additionalProperties": false,
       "properties": {
+        "pagination": {
+          "type": "object",
+          "properties": {
+            "max-page-size": {
+              "type": "integer",
```
Comment (seantleonard) [inline schemas/dab.draft.schema.json:146]:
```suggestion
              "type": [
                "integer",
                "null"
              ],
```

File: schemas/dab.draft.schema.json [hunk 4]
```diff
@@ -139,6 +139,23 @@
...
+              "description": "Defines the maximum number of records that can be returned in a single page of results. If set to null, the default value is 100,000.",
+              "default": 100000,
+              "minimum": 1
+            },
+            "default-page-size": {
+              "type": "integer",
```
Comment (seantleonard) [inline schemas/dab.draft.schema.json:152]:
```suggestion
              "type": [
                "integer",
                "null"
              ],
```

**Review by seantleonard — CHANGES_REQUESTED** [review]
needs to update to allow null for properties that are optional. That way `dab validate` won't fail to validate config files that exclude those properties and prefer DAB defaults.

File: schemas/dab.draft.schema.json [hunk 5]
```diff
@@ -207,6 +233,16 @@
           "description": "Global hosting configuration",
           "additionalProperties": false,
           "properties": {
+            "max-response-size-mb": {
+             "type": [
+                "integer",
+                "null"
+              ]
+              "description": "Specifies the maximum size, in megabytes, of the database response allowed in a single result. If set to null, the default value is 158 MB.",
```
Comment (Aniruddh25) [inline schemas/dab.draft.schema.json:241]: Why is default value = 158 MB? Why not some power of 2 for e.g. 128 or 256?

**Review by Aniruddh25 — APPROVED** [review]
LGTM, just one question out of curiosity.

File: schemas/dab.draft.schema.json (hunk 5 above)
Comment (seantleonard) [reply schemas/dab.draft.schema.json:241]: See #2242:
https://github.com/Azure/data-api-builder/blob/f97aa96b70371ea00ea9d320b6deecde1ebde0c5/src/Config/ObjectModel/HostOptions.cs#L11-L21

**Review by seantleonard — APPROVED** [review]

**Review by seantleonard — CHANGES_REQUESTED** [review]
actually, there's still issues in unit test failure. Need to investigate.

File: schemas/dab.draft.schema.json [hunk 6]
```diff
@@ -207,6 +233,16 @@
           "description": "Global hosting configuration",
           "additionalProperties": false,
           "properties": {
+            "max-response-size-mb": {
+             "type": [
+                "integer",
+                "null"
+              ]
```
Comment (seantleonard) [inline schemas/dab.draft.schema.json:240]:
```suggestion
              ],
```

**Review by Aniruddh25 — APPROVED** [review]

**Review by seantleonard — APPROVED** [review]

**Pull request merged**